support/packages/github-to-jira-synchronization-tool/src/config/config.js

```diff
@@ -3,21 +3,45 @@
  * SPDX-License-Identifier: MIT
  */
 
+const fs = require('fs');
+const path = require('path');
+
 let config = null;
+let lastModifiedTime = 0;
+
+const CONFIGURATION_PATH = path.resolve(__dirname, '../../mapping-config.json');
+
+function fileHasChanged() {
+	const stat = fs.statSync(CONFIGURATION_PATH);
+
+	if (stat.mtime > lastModifiedTime) {
+		lastModifiedTime = stat.mtime;
+
+		return true;
+	}
+
+	return false;
+}
 
 function loadConfig() {
-	if (config) {
+	if (!fileHasChanged() && config) {
 		return config;
 	}
 
 	try {
-		config = require('../../mapping-config.json');
+		config = JSON.parse(fs.readFileSync(CONFIGURATION_PATH, 'utf-8'));
 
 		return config;
 	}
 	catch (error) {
+		if (config) {
+			console.error('new config file is wrong, reusing past config');
```

Comment:
rock solid :+1:

Reply:
😂  I didn't want the tool to fail when hot-changing the configuration. Once the tool starts it checks the config and fails if the config is correct, but if we at least one valid configuration is better to stick to that instead of aborting I think 🤔  what do you think?

Reply:
No, no, I like it. I found it a very good idea +1

As long as the user gets immediate feedback if he changed the config and the load went wrong it's good. Otherwise (if the user won't be clearly notified) it may be a cause of trouble if the user looks at the config and think it's right but the application does a different thing.

Reply:
Actually I realized yesterday that with console.error they won't get immedita feedback, going to use log here to put this message in the log file